components/script/dom/location.rs

```diff
@@ -57,7 +60,10 @@ impl LocationMethods for Location {
 
     // https://html.spec.whatwg.org/multipage/#dom-location-reload
     fn Reload(&self) {
-        self.window.load_url(self.get_url(), true, None);
+        self.window.main_thread_script_chan().send(
+            MainThreadScriptMsg::Navigate(self.window.pipeline_id(),
+                                          LoadData::new(self.get_url(), None, Some(self.get_url())),
+                                          true)).unwrap();
```

Comment:
My previous plan is: for navigate request that should always happen, just send the message to `MainThreadScriptMsg`; otherwise call `window.load_url()`. But I bet you have a reason for this since I have no idea about the character of `ScriptThread`.